servers/rendering/renderer_rd/shader_compiler_rd.cpp

```diff
@@ -1197,6 +1198,7 @@ String ShaderCompilerRD::_dump_node_code(const SL::Node *p_node, int p_level, Ge
 							const SL::VariableNode *varnode = static_cast<const SL::VariableNode *>(onode->arguments[i]);
 
 							StringName texture_uniform = varnode->name;
+							is_screen_texture = (texture_uniform == "SCREEN_TEXTURE");
```

Comment:
All this mechanism most likely needs some enabling in ShaderCompilerRD because, as an example, in 2D or in the clustered rendering, you dont want to use it.

Reply:
Yeah that makes sense, I'm still learning the parser code so not sure what would be the easiest way to switch on it? Can I check if this is a spatial shader?

Reply:
@reduz I added a `apply_luminance_multiplier` boolean to `DefaultIdentifierActions` that is only set by the shader compiler for the mobile renderer. Not sure if this is what you had in mind but it seems to do the trick :)